### ISSUE TITLE
oxfmt: `--list-different` throws error in terminal

### ISSUE THREAD

**Issue by ivodolenc** [oxc-project/oxc#19776]
Hi, sorry if this isn’t a bug, I’m looking into migrating from Prettier and using this `oxfmt` config for the first time, so it’s possible I missed something.

> Note: Im using pnpm workspace (monorepo) and I installed `oxfmt` according to official docs.

I'd like to see a list of formatted files (same as `prettier` logs), and currently when I use this config in `package.json` scripts:

> [--list-different](https://oxc.rs/docs/guide/usage/formatter/cli.html#output-options) — List files that would be changed

```json
{
  "scripts": {
    "format:pkg": "oxfmt packages/pkg --list-different"
  }
}
```

Run cmd:

```sh
pnpm format:pkg
```

It starts formatting, logs first changed file, but then throws an error and immediately stops:

```txt
packages/pkg/src/index.ts ELIFECYCLE  Command failed with exit code 1.
```

If I remove `--list-different` arg:

```json
{
  "scripts": {
    "format:pkg": "oxfmt packages/pkg"
  }
}
```

It works as expected and doesn't log anything and no errors.

**Comment by leaysgur** [comment]
Thank you for reporting this!

However, this is actually expected behavior.

`--list-different` exits with code `1` when there are files that differ from the expected formatting. And I think this is the same for Prettier.

That message you're seeing is pnpm interpreting the non-zero exit code as an error.

**Comment by ivodolenc** [comment]
Ah, my bad, thanks for the quick info.

I was expecting logs of all formated files just like `prettier` outputs file paths when we run `prettier --write .`:

```txt
.github/release.yml 29ms (unchanged)
.vscode/extensions.json 31ms (unchanged)
.vscode/settings.json 2ms (unchanged)
bundler.config.ts 58ms (unchanged)
CHANGELOG.md 28ms (unchanged)s
eslint.config.js 6ms (unchanged)
package.json 3ms (unchanged)
...
```

Is this possible to configure in `oxfmt` with some other flags or options?

Or maybe just logs formatted paths if there are any?

This can be useful for quickly scanning formatted paths and gives me some sense of confirmation that the process is done, rather than no records or messages.

**Comment by ivodolenc** [comment]
In monorepo, at the moment, it's impossible to see which files are formatted without filepath logs. 

If i specify direct path for `oxfmt packages/pkg` dir, i dont know if the whole monorepo is formatted or just this specific dir since i dont see any logs.

When running `oxfmt --list-different`, this just logs filepath and quits after first match that will be changed, but actually doesnt format file, which is very confusing, what this flag do? Why quits after first match? Isnt this soupose to find all files that will be changed, format the files and logs them out? 

Also usage with pnpm it constantly throws `ELIFECYCLE  Command failed with exit code 1.` at the end of commands process. With `npm` doesnt throw any error so maybe this is a `pnpm` bug.

Sorry about the noise, this is just personal opinion and feedback, im used to prettier behavior.

**Comment by leaysgur** [comment]
For your use case, here are two workarounds:

Check before formatting:

```sh
oxfmt packages/pkg --list-different || true
```

Check what was formatted after the fact:

```sh
oxfmt packages/pkg && git diff --name-only
```

The reason oxfmt stays quiet by default is that verbose file listings rarely help in practice — they tend to flood the terminal (which has scroll limits), and tools like git diff can serve the same purpose more reliably.

Still if you’d like built-in support for this, feel free to open a feature request.

**Comment by ivodolenc** [comment]
Thanks for workarounds, I guess I just need to get used to the new system.

Also, I just found some issue where you note that `oxfmt` is being switched to quiet mode since version `0.14+`. True, prettier logs unchanged and changed paths, but why not just logs formatted (changed) paths, if any?

Or add a new flag that enables only formatted logs, or some indication for the case where we specify a custom path for a specific directory so I know that only this directory will be formatted (for monorepos).

Anyway, just thinking out loud, thanks for the quick help, I appreciate it.
